docs/source/azure/policy/resources/keyvault.rst

```diff
@@ -99,4 +105,24 @@ This policy will find all KeyVaults with an access of Service Principals not in
                   secrets:
                     - get
                   certificates:
-                    - get
\ No newline at end of file
+                    - get
+
+This policy will find all KeyVaults and add get and list permissions for keys.
+
+.. code-block:: yaml
+
+    policies:
+        - name: policy
+          description:
+            Add get and list permissions to keys
+          resource: azure.keyvault
+          actions:
+            - type: access-policy
+              operation: add
+              access_policies:
+                - tenant_id: 00000000-0000-0000-0000-000000000000
```

Comment:
I think `access-policy` action brings more value if it allows a few additional things:

1. Human readable Service Principal names, not only `object_id`. We can translate SP->object id using Graph API
2. Allow extend\replace\revoke access policies for a group of Principals without access level duplications.
3. I can imagine 3 operations, performed by the action:
    - Extend: adds specified access level to a list of people, if they have higher or equal access level -- no changes
    - Replace: enforces specified access level to a list of people
    - Revoke: removes specified access levels from everyone, except for the specified list of people (aka admins)

It can be different actions or a single action with `operation` parameter.

Reply:
spoke offline with @aluong and @logachev and we decided to keep v1 of this action simple and mirror the experience of adding/updating access policies with the Azure portal. if there is enough interest, we will add functionality around servicePrincipal names and object ids.

Reply:
Well to be fair the portal does not make you use object ids :)  

Would be nice to think about some global support for this lookups that spans all filters/actions.   (For when you are providing an ID in your policy)